pipeline.py

```diff
@@ -1,83 +1,125 @@
 """
 Script for analysing which areas of the brain are activated when identifying different pictures.
 NOTE: This script is an exercise version and should NOT be used in actual analysis.
-STATUS: Currently DRAFT for analysing a single subject.
 """
 
 import nibabel as nib
-import numpy as np
 import pandas as pd
 from nilearn import plotting
 from tools import *  # make sure namespace doesn't get messed up!
 from matplotlib.pyplot import matshow, savefig
 
+
 # Input params
 searchlight_radius = 2
 
-# Load original data
-bold = nib.load('subj1/bold.nii.gz') # loads dataset of BOLD images in Nifti1Image format
-mask = nib.load('subj1/mask4_vt.nii.gz')
-anat = nib.load('subj1/anat.nii.gz')
-labels = pd.read_csv('subj1/labels.txt', sep=' ')
+subject_list = [1, 2, 3, 4, 5, 6]  # number identificators for test subjects to be used in analysis
```

Comment:
So, is the official way to run this code for a specific subject to go in your code and make a change like this?

```
# subject_list = [1, 2, 3, 4, 5, 6]  # number identificators for test subjects to be used in analysis
subject_list = [4]
```

This is certainly standard practice for many analysis pipelines I have seen. But can we do better? In the analysis script you write for your real studies, do you re-run the whole pipeline for all subjects every time you make a change to the code? Do you have ways to run just parts of it? If so, what kind of solutions have you come up with? I'm curious :)

Reply:
This is a bit late but I'm not sure what would 'running just a part of the script' mean here. If data changes, everything must be re-run. If code changes, unless it at the last bits, everything must also be re-run because previous steps feed the next ones. True, here the model RDM could be separated into a separate script, with results saved into a file. 

Regarding the subject selection, adding that as a command line argument could be better. Let's see if I have enough time to implement.